test/ui/QuerySummaryTest.ts

```diff
@@ -156,5 +202,121 @@ export function QuerySummaryTest() {
         expect($$(test.cmp.element).text()).not.toEqual(jasmine.stringMatching(/^Results.*of.*/));
       });
     });
+
+    describe('when the option enableNoResultsFoundMessage is set to true', () => {
+      it(`when the noResultsFoundMessage is not specified
+          it should return the default message`, () => {
+        test = Mock.optionsComponentSetup<QuerySummary, IQuerySummaryOptions>(QuerySummary, {
+          enableNoResultsFoundMessage: true
+        });
+        expect(test.cmp.options.noResultsFoundMessage).toBe(`No results for ${queryTag}`);
+      });
+
+      it(`when the noResultsFoundMessage is specified 
+          it should return the custom message`, () => {
+        test = Mock.optionsComponentSetup<QuerySummary, IQuerySummaryOptions>(QuerySummary, {
+          enableNoResultsFoundMessage: true,
+          noResultsFoundMessage: 'customMessage'
+        });
+        expect(test.cmp.options.noResultsFoundMessage).toBe('customMessage');
+      });
+
+      describe('when the noResultsFoundMessage is parsed', () => {
+        it(`when there is no query tag in the noResultsFoundMessage
+            it should not change the initial message`, () => {
+          test = Mock.optionsComponentSetup<QuerySummary, IQuerySummaryOptions>(QuerySummary, {
+            enableNoResultsFoundMessage: true,
+            noResultsFoundMessage: 'customMessage'
+          });
+
+          test.env.queryStateModel.get = () => 'querySearched';
+          Simulate.query(test.env, { results: FakeResults.createFakeResults(0) });
+
+          const customMessageElement = $$(test.cmp.element).find('.coveo-query-summary-no-results-string');
+
+          expect(customMessageElement.textContent).toBe('customMessage');
+        });
+
+        it(`when there is one query tag in the noResultsFoundMessage
+            it should replace the query tag by the query searched`, () => {
+          test = Mock.optionsComponentSetup<QuerySummary, IQuerySummaryOptions>(QuerySummary, {
+            enableNoResultsFoundMessage: true,
+            noResultsFoundMessage: `customMessage ${queryTag}`
+          });
+
+          test.env.queryStateModel.get = () => 'querySearched';
+          Simulate.query(test.env, { results: FakeResults.createFakeResults(0) });
+
+          const customMessageElement = $$(test.cmp.element).find('.coveo-query-summary-no-results-string');
+
+          expect(customMessageElement.textContent).toBe('customMessage querySearched');
+        });
+
+        it(`when there is multiple query tags in the noResultsFoundMessage
```

Comment:
I'd add a test for XSS here also (kinda like you did with the <script> tag for the custom no results element).

I'd add a test if the `querySearched` is an empty string.

I'd add a test if somebody searches for `${query}`, to see how does the parsing logic handles it.

Reply:
Good point 👍  !

- new XSS test : The queryTag replacement mechanism in the custom message and in the no results found page is the same. The queryTag will be replaced by a queryTagContainer in both scenarios. This is why I added only one test that test if the content of this container is escaped properly. 

- Empty querySearched test : For the test about the querySearched when it is an empty string I tested that « even if the query search is empty, the query tags will be properly replaced by an empty string. I did not consider the fact that the no results page will no be shown when the query searched is empty because I consider this logic not specific to this component. Should it be considered?  

- No problem!